CMakeLists.txt

```diff
@@ -333,6 +333,20 @@ function(generateInstallDirectives)
     )
 
   elseif(PLATFORM_WINDOWS)
+
+    # CMake doesn't prefer 'Program Files' on x64 platforms, more information
+    # here: https://gitlab.kitware.com/cmake/cmake/-/issues/18312
+    # This is a workaround, to ensure that we leverage 'Program Files' when
+    # on a 64 bit system.
+    if(WIN32)
+      if(CMAKE_SIZEOF_VOID_P MATCHES "8")
+          set(CMAKE_INSTALL_PREFIX "Program Files/${CMAKE_PROJECT_NAME}")
+      else()
+          set(CMAKE_INSTALL_PREFIX "Program Files (x86)/${CMAKE_PROJECT_NAME}")
```

Comment:
If this is what gets chocolatey going, this have to be differentiated from the WiX/MSI path, because the CMAKE_INSTALL_PREFIX normally controls where the `install` command copies the files, and so for CPack it often only affects the place where files are copied in the intermediary directory which is then used by the real package tool to pick up the files.

I guess it depends if the package acts as an archive of sorts, or it has custom installation behavior.

Also, nitpick, `if(WIN32)` is redundant.

Reply:
Actually, I realized now that the install is used to provide the files to the other logic of CMake (I was still thinking of the install calls used directly by CPack, which is not the case).
So indeed, it's the osquery-packaging repository that needs to switch the folders from where it takes the files, depending on the bitness, here:  https://github.com/osquery/osquery-packaging/blob/ab1a6e6f5f98186b3144fd4d5476dca5b7081fa6/platform/windows/common.cmake#L29

And there's already a cache variable that can be used: https://github.com/osquery/osquery-packaging/blob/ab1a6e6f5f98186b3144fd4d5476dca5b7081fa6/platform/windows/common.cmake#L11

Reply:
I don't think this is the right approach, but I also haven't spent a lot of time deep-diving.

I think the root cause is that during the archive creation process @Smjert describes, CPack is creating a relative `./Program Files (x86)` directory for `install` targets with the `PROGRAM` argument. We need to look at what influences this subdir selection and set the relevant (I hope) environment variables.

Reply:
So I tested and CMake needs a bit more convincing right now to set that prefix, because it's in the cache and it's set before executing any other CMake code.
There's also a second issue, which is that the path should start with `/` otherwise it won't be appended to the DESTDIR correctly.

So this should become
```cmake
 if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
     if(CMAKE_SIZEOF_VOID_P MATCHES "8")
         set(CMAKE_INSTALL_PREFIX "/Program Files/${CMAKE_PROJECT_NAME}" CACHE PATH "" FORCE)
     else()
         set(CMAKE_INSTALL_PREFIX "/Program Files (x86)/${CMAKE_PROJECT_NAME}" CACHE PATH "" FORCE)
     endif()
 endif()
```

Basically this sets the flag only if a user has not changed the `CMAKE_INSTALL_PREFIX` manually via CLI.